Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -317,16 +317,6 @@ WebProcessPool::~WebProcessPool()
 
         process->shutDown();
     }
-
-    if (processPools().isEmpty()) {
```

Comment:
Technically network process is held alive by WKWebsiteDataStore, not WKWebView.
To solve my test failure, we can try holding a strong ref to NetworkProcessProxy when WebsiteDataStore is waiting for network process's reply.
But it seems wrong to terminate network process when there's no web process using it, because network process carries in-memory states (e.g. IDB data for non-persistent sessions). Relaunching network process will lead to data loss for the WebsiteDataStore.

Reply:
The default WebsiteDataStore has no reason to stay alive here since there is no WKWebView using it (anymore).

Again, for many apps, they merely use WKWebView for something minimal (e.g. showing a help page). It would not be acceptable for them to carry a network process forever simply because they once displayed a help page.